src/hotspot/share/opto/regmask.hpp

```diff
@@ -213,16 +360,30 @@ class RegMask {
   // Verify watermarks are sane, i.e., within bounds and that no
   // register words below or above the watermarks have bits set.
   bool valid_watermarks() const {
```

Comment:
For sanity, should we also assert here, and enforce across the code, that `_lwm <= _hwm`, even in the case of `Size() == 0` or is this invariant too expensive/difficult to maintain?

Reply:
We could do that, but I'm not sure I see the benefit. Could you elaborate a bit?

Reply:
The benefit, in my opinion, is better comprehensibility due to a simpler model with stronger invariants. An alternative would be to extend the comment at the definition of `_lwm` and `_hwm` clarifying that the value of these variables is unspecified when the register set is empty.

Reply:
Recapping what we discussed in-person here: we should not try to enforce `_lwm <= _hwm`, but describe how they work and what they guarantee in more detail through comments in `regmask.hpp`.